### PR TITLE
ROS independence

### PR DESCRIPTION
**Pull request by stefanwapnick**
Moved ROS specific code to own implementation files, referenced by main node and algorithm node, making it independent of specific ROS version.

### CODE REVIEW

File: src/perception/avt_341_perception_node.cpp [hunk 1]
```diff
@@ -108,78 +104,48 @@ int main(int argc, char *argv[]) {
...
-	}
-	if (ros::param::has("~overhead_clearance")){
-		ros::param::get("~overhead_clearance",overhead_clearance);
-	}
+    n->get_parameter("~grid_res", grid_res, 1.0f);
+    n->get_parameter("~grid_llx", grid_llx, -100.0f);
+    n->get_parameter("~grid_lly", grid_lly, -100.0f);
+
+    n->get_parameter("~warmup_time", warmup_time, 1.0f);
+    n->get_parameter("~slope_threshold", thresh, 1.0f);
```
Comment (stefanwapnick) [inline src/perception/avt_341_perception_node.cpp:125]: This used to be ~threshold in the previous version but I noticed that in the launch file it is defined as slope_threshold so I changed it to this.

Comment (stefanwapnick) [inline src/perception/avt_341_perception_node.cpp:125]: This used to be ~threshold in the previous version but I noticed in the launch file that it was defined as ~slope_threshold so I changed it here.

**Comment by MahyarBayran-Naisense** [comment]
- In CMakeLists, for ros2 build, catkin is used which should be changed. 
- Is PCL needed for ros2 config?
- I don't see ament_cmake support. Maybe that's something that is missing?

```
if($ENV{ROS_DISTRO} STREQUAL "noetic")
  find_package(catkin REQUIRED COMPONENTS
    roscpp
    rospy
    pcl_ros
    std_msgs
    tf
  )
else()
 find_package(ament_cmake REQUIRED COMPONENTS
    rclcpp
    rclpy
    std_msgs
    tf2_ros
  )
endif()
```

It also seems like CMakeList

**Comment by CGoodin** [comment]
Looks good to me. I'm able to build and run the code with the same result as before the conversion. I have not tested with ROS2.

**Comment by stefanwapnick** [comment]
@MahyarBayran-Naisense @CGoodin @DanCarruth @wang686503
Just a quick update based on Mahyar's review: I moved the pcl library to a seperate file so that the ros node code does not directly depend on it. On the ROS2 branch (which is more focused on Windows OS), since the pcl library is only used in one place and it is a bit more annoying to install on Windows, we just do the point cloud generation manually instead of using the PCL library. By putting the PCL library in a separate file it should make the sim node more easily portable to the ROS2 branch without change. 

I wanted to do something similar with X11 also: In the ROS2 branch we removed XLib visualization and replaced it with OpenCV and RVIZ (there is a flag to switch between OpenCV or RVIZ visualisation of planned paths) since we had trouble with X11 Windows installation. 

I think in the future I would petition to remove X11 and replace it with OpenCV and/or RVIZ so that it could be more portable between OS, but I am not entirely familiar with X11 so I am not sure if it somehow provides more flexibility than just drawing on an OpenCV image. 

But I guess this is maybe a discussion for later and we can do this in a separate pull request (if approved). 
Unless you all agree that you would like to immediately remove X11 for OpenCV and/or RVIZ, in which case we have a working version on the ROS2 branch so I could just copy it over to this pull request.

**Comment by stefanwapnick** [comment]
@CGoodin
Sounds good. So would you be open for another pull request to replace X11 with OpenCV + RVIZ immediately after this one, or do you want to discuss it more?

**Comment by CGoodin** [comment]
I would like to discuss replacing X11 at our next meeting so I can understand the implications.

**Pull request merged**